internal/consumer/partitionConsumer.go

```diff
@@ -243,6 +243,13 @@ func (p *partitionConsumer) messageLoop(offsetRange *kafka.OffsetRange) {
 			p.tally.Gauge(metrics.KafkaPartitionTimeLag).Update(float64(lag))
 			p.tally.Gauge(metrics.KafkaPartitionReadOffset).Update(float64(m.Offset))
 			p.tally.Counter(metrics.KafkaPartitionMessagesIn).Inc(1)
+			delay := p.topicPartition.RetryDelay
```

Comment:
We should probably move this to before the metrics are set. If Delay is long, it will give user incorrect impression that the message has been processed (e.g., KafkaPartitionReadOffset, which denotes the offset of the latest message that the consumer has consumed, metric might be incremented even though the message has not been delivered to user until sleep finish).

Reply:
good catch, will change it correspondingly.